draft-ietf-taps-arch.md

```diff
@@ -402,7 +403,7 @@ This list of events that can be delivered to an application is not exhaustive, b
 
 The Transport System Implementation Concepts define the set of objects used internally to a system or library to implement the functionality needed to provide a transport service across a network, as required by the abstract interface.
 
-* Connection Group: A set of Connections that share properties. For multiplexing transport protocols, the Connection Group defines the set of Connections that can be multiplexed together.
+* Connection Group: A set of Connections that share properties. For multiplexing transport protocols, the Connection Group defines the set of Connections that can be multiplexed together. Groups can be defined implicitly by the Implementation, based on which properties are compatible. However, applications SHOULD be able to explicitly define Connection Groups, as discussed in {{groups}}.
 
```

Comment:
Defining the Connection Groups implicitly is a bit different from what we have in the API draft at the moment where I would read it as Connection Groups are formed explicitly by calling Clone, So the API text should probbaly be aligned a bit if we change it here. 

Related to this, how do I know as an application what connections I affect when changing a protocol property if connections are grouped implicitly? We have in the API draft: 

> Connections in a Connection Group share all Protocol Properties that are not applicable to a Message. Changing one of these Protocol Properties on one Connection in the group changes it for all others. For example, changing "Timeout for aborting Connection" (see {{conn-timeout}}) on one Connection in a group will automatically change this Protocol Property for all Connections in the group in the same way. 

I think we would need some comment about the effect of the implicit grouping in the text above. With implicit grouping changing a protocol property may affect multiple connectione even if they were not explicitly grouped with Clone. This may not be a problem, but the semantics should be clear.

Reply:
I also wanted such implicit grouping some time ago, but @abrunstrom convinced me that this doesn't add value (and indeed creates problems like she describes here). That's because "clone" only expresses a wish for Connections to be grouped in some form, but there is no promise that much is really happening under the hood. I.e., after calling "clone", an application doesn't really know if its Connections WILL be multiplexed, and if priorities between them WILL play out correctly. However, there's a consistent behavior now regarding the things that ARE promised - e.g. the Connection Properties that she describes here: if one is changed, it affects all the Connections within a Group.

As a conclusion, rather than changing the text she quotes, I think this implicit grouping should be removed. The default behavior for an application should just be to use Clone whenever possible ... whenever there's no good reason NOT to use it.